File: selection_sort.py
#Selection Sort
#Done by mohammed zubair
#Date: 15-09-2020
def selection_sort(array,n):
    mini=0
    temp=0
    for i in range(0,n-1):
        mini=i
        for j in range(i+1,n):
            if(array[j]<array[mini]):
                mini=j
        temp=array[i]
        array[i]=array[mini]
        array[mini]=temp
    print("\nSorted List...:")
    for i in array:
        print(i,end=" ")
        
        
        
    
n=int(input("Please Enter the number of elemnts in list:"))
l=[]
print("\nEnter the list elements:")
for i in range(0,n):
    z=int(input())
    l.append(z)
selection_sort(l,n)





    

File: selection_sort.c
/* Selection Sort
Date: 15-09-2020*/
#include<stdio.h>
#include<stdlib.h>

void SelectionSort(int *list,int n );
int main()
{
	int i,n,*list;
	printf("Please Enter the number of elements in list:");
	scanf("%d",&n);
	list=(int*)malloc(n*sizeof(int));
	printf("\nEnter the elements of list....\n");
	for(i=0;i<n;i++)
	{
         scanf("%d",&list[i]);
  	}
  	SelectionSort(list,n);
  	return 0;
}

void SelectionSort(int *list,int n)
{

	int temp=0,min,i,j;
	for(i=0;i<n-1;i++)
	{
		min=i;
		for(j=i+1;j<n;j++)
		{
			if(list[j]<list[min])
				min=j;

		}
		temp=list[i];
		list[i]=list[min];
		list[min]=temp;
	}
	printf("Sorted List...\n");
	for(i=0;i<n;i++)
		printf("%d\t",list[i]);
}
